src/lexer/tokens.js

```diff
@@ -32,6 +32,21 @@ module.exports = {
       }
     }
 
+    if (id === this.tok.T_ENUM) {
+      if (this.version < 801) {
+        return this.tok.T_STRING;
+      }
+      const initial = this.offset;
+      let ch = this.input();
+      while (ch == " ") {
+        ch = this.input();
+      }
+
+      const is_enum = this.is_LABEL_START();
+      this.unput(this.offset - initial);
```

Comment:
I agree, this is much nicer! Thinking of how the PHP lexer handles this, I think this misses an edge case though. I think you need to look ahead through the entire label to check if it's not equal to `implements` or `extends`, to handle:

```php
class Enum extends Foo {}
class Enum implements Bar {}
```

Reply:
Good call, I totally missed that! Fixed.